datastore/query.go

```diff
@@ -176,6 +326,22 @@ func (q *Query) Transaction(t *Transaction) *Query {
 	return q
 }
 
+// FilterEntity returns a query with provided filter.
+//
+// Filter can be a single field comparison or a composite filter
+// AND and OR are supported composite filters
+// Filters in multiple calls are AND'ed together
+func (q *Query) FilterEntity(entityFilter EntityFilter) *Query {
+	q = q.clone()
+	validEntityFilter, err := entityFilter.toEntityFilter()
```

Comment:
quibble: I would shorten `validEntityFilter`. This is a pretty long name for a variable that isn't used outside of this method. I recommend using something like `vf`.

Reply:
Done